addons/website_sale/controllers/main.py

```diff
@@ -983,7 +983,7 @@ def shop_checkout(self, try_skip_step=None, **query_params):
                     order_sudo._set_delivery_method(delivery_method, rate=rate)
 
         if try_skip_step and can_skip_delivery:
-            return request.redirect('/shop/confirm_order')
+            return self._redirect_to_next_checkout_step()
 
         checkout_page_values.update(
             request.website._get_checkout_step_values()
```

Comment:
It's better to use our new method to get dynamically the next checkout step. It will automatically handle enabled, disabled, and/or custom steps. Don't forget to also apply this comment to your enterprise PR.
```suggestion
        checkout_page_values.update(
            request.website._get_checkout_step_values()
        )
        if try_skip_step and can_skip_delivery:
            return checkout_page_values['next_website_checkout_step_href']
```